library/nametab.ml

```diff
@@ -582,3 +582,28 @@ let global_inductive qid =
   | ref ->
       user_err ?loc:qid.CAst.loc
         (pr_qualid qid ++ spc () ++ str "is not an inductive type.")
+
+type modules_frozen_t = MPTab.t * MPDTab.t * DirTab.t * mprevtab * mptrevtab
```

Comment:
Is it possible to group all this stuff in a single `ref`? The use of `Summary.ref` is clearly indicating a pattern in the code while here it's a bit drowned in the noise. (This is not unheard of, but ideally we should stop manipulating the state by hand and only rely on the combinators if we want to move to a finer implementation.)

Reply:
Sure! Do you want a record with mutable fields, or a ref with functional updates?

Reply:
The latter.

Reply:
Here you go!